src/client/browser/idle-page/status-indicator.js

```diff
@@ -0,0 +1,234 @@
+const SPINNER_WIDTH        = 8;
+const RELATED_SPINNER_SIZE = 0.8;
+const MAXIMUM_SPINNER_SIZE = 400;
+const ROTATION_ANGLE       = 7;
+
+const LINE_HEIGHT_INDENT = 6;
+
+const CONNECTED_SPINNER_COLOR    = '#28687F';
+const DISCONNECTED_SPINNER_COLOR = '#882E24';
+const SPINNER_GRADIENT_COLOR     = '#181818';
+
+const CONNECTED_TEXT    = 'CONNECTED';
+const DISCONNECTED_TEXT = 'DISCONNECTED';
+
+const PAGE_BACKGROUND_CLASS_NAME    = 'page-background';
+const CONTAINER_CLASS_NAME          = 'container';
+const STATUS_TEXT_AREA_CLASS_NAME   = 'status-text-area';
+const USER_AGENT_ELEMENT_CLASS_NAME = 'user-agent';
+const STATUS_ELEMENT_CLASS_NAME     = 'status';
+const CANVAS_CLASS_NAME             = 'spinner';
+
+const CONNECTED_CLASS_NAME    = 'connected';
+const DISCONNECTED_CLASS_NAME = 'disconnected';
+
+const ANIMATION_DELAY = 30;
+
+
+//Utils
+function convertToRadian (angle) {
+    return angle * Math.PI / 180;
+}
+
+function convertToString (value) {
+    return value + 'px';
+}
+
+function rotateAxes (point, rotationAngle) {
+    var angle = convertToRadian(rotationAngle);
+
+    return {
+        x: Math.round(point.x * Math.cos(angle) - point.y * Math.sin(angle)),
+        y: Math.round(point.x * Math.sin(angle) + point.y * Math.cos(angle))
+    }
+}
+
+function moveAxes (point, distance) {
+    return {
+        x: Math.round(point.x - distance),
+        y: Math.round(point.y - distance)
+    }
+
+}
+
+export default class StatusIndicator {
+    constructor () {
+        this.connected     = true;
+        this.canvas        = document.getElementsByClassName(CANVAS_CLASS_NAME)[0];
+        this.canvasContext = this.canvas.getContext("2d");
+
+        this.spinnerAnimationInterval = null;
+        this.rotationAngle            = 0;
+        this.size                     = null;
+        this.spinnerCenter            = null;
+        this.gradient                 = null;
+
+        this._setSize();
+        this._setFontSize();
+        this._setSpinnerGradient();
+
+        StatusIndicator._createStatusMessage(this.connected);
+        StatusIndicator._alignContainerVertically();
+
+        this._drawSpinner(this.connected, 0);
+
+        window.onresize = (() => {
```

Comment:
Refactor this to the separate method, something like `_watchWindowResize` to not pollute ctor with business logic